libvast/vast/format/json.hpp

```diff
@@ -156,7 +156,20 @@ caf::error reader<Selector>::read_impl(size_t max_events, size_t max_slice_size,
   VAST_ASSERT(max_events > 0);
   VAST_ASSERT(max_slice_size > 0);
   size_t produced = 0;
-  for (; produced < max_events; lines_->next()) {
+  table_slice_builder_ptr bptr = nullptr;
+  bool timeout = false;
+  auto next_line = [&] {
+    if (!bptr || bptr->rows() == 0)
```

Comment:
Maybe we should add a warning here why we need the special case for `rows() == 0`, since it apparently wasnt obvious from reading the same logic in the zeek reader.

Reply:
That's a fair point, but I think since we plan to rework this in the near future anyways with a central place for input management, this does not need to be done.